### PR TITLE
Disable keys and logins in unrecognised users job

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
This adds password removal and access key disablement to the new unrecognised users job. It also sends a disablement email notification. 

## What is the value of this?
Fleshes out more of the functionality of the job before we hook it up to real data and do some end to end testing.

## Will this require CloudFormation and/or updates to the AWS StackSet?
No

## Any additional notes?
No

### CODE REVIEW

File: hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala [hunk 1]
```diff
@@ -1,33 +1,71 @@
...
+      None
+    )
+
+    val janusUsers: Seq[String] = getJanusUsernames(dummyJanusData)
+    val allCredsReports = cacheService.getAllCredentials
+
+    allCredsReports.map { case (account, eitherCredsReportOrFailure) =>
+      eitherCredsReportOrFailure.map { credsReport =>
+        val humanUsers = credsReport.humanUsers
+        val unrecognisedIamUsers: Seq[VulnerableUser] = filterUnrecognisedIamUsers(humanUsers, "name", janusUsers)
```
Comment (jorgeazevedo) [inline hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala:51]: The interface for `filterUnrecognisedIamUsers` opens up the possibility for the method to be called with different `targetKey` values. Is that the case? If not I'm wondering if it should be a constant in `IamUnrecognisedUsers`

Comment (jfsoul) [reply hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala:51]: Absolutely, we discussed that but didn't go back and change it. Maybe we could address in #309 where it's introduced

File: hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala [hunk 2]
```diff
@@ -1,33 +1,71 @@
...
 
-  //TODO disable access key and remove password
-  //TODO send notification that this has been done
+        if(unrecognisedIamUsers.nonEmpty) {
```
Comment (jorgeazevedo) [inline hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala:53]: Question on scala best practice. What's the party line on side effects from a map? Should the  bit of code that runs when this is true be in a separate forEach?

**Review by jorgeazevedo — APPROVED** [review]

File: hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala (hunk 2 above)
Comment (jfsoul) [reply hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala:53]: Party line is use foreach 👍 

There's probably a valuable refactor to give some of these methods return types and centralise the error handling, but I think that can come later and applies to the other scheduled job too

File: hq/test/schedule/IamUnrecognisedUserTest.scala [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
-import model.{AccessKey, CredentialReportDisplay, Green, HumanUser, NoKey, Tag, AwsAccount => Account}
+import model._
 import org.joda.time.{DateTime, Seconds}
 import org.scalatest.{FreeSpec, Matchers}
-import schedule.unrecognised.IamUnrecognisedUsers.{filterUnrecognisedIamUsers, getHumanUsers, getJanusUsernames}
-import utils.attempt.FailedAttempt
+import schedule.unrecognised.IamUnrecognisedUsers.{USERNAME_TAG_KEY, filterUnrecognisedIamUsers, getJanusUsernames}
 
 class IamUnrecognisedUserTest extends FreeSpec with Matchers {
   val humanUser1 = HumanUser("", true, AccessKey(NoKey, None), AccessKey(NoKey, None), Green, None, None, List(Tag("name", "ade.bimbola")))
```
Comment (Nirvikalpa108) [inline hq/test/schedule/IamUnrecognisedUserTest.scala:10]:
```suggestion
  val humanUser1 = HumanUser("", true, AccessKey(NoKey, None), AccessKey(NoKey, None), Green, None, None, List(Tag(USERNAME_TAG_KEY, "ade.bimbola")))
```

File: hq/app/schedule/unrecognised/IamUnrecognisedUsers.scala [hunk 4]
```diff
@@ -1,22 +1,20 @@
...
 
-  def filterUnrecognisedIamUsers(iamHumanUsersWithTargetTag: List[HumanUser], targetKey: String, janusUsernames: List[String]): List[HumanUser] =
+  def filterUnrecognisedIamUsers(iamHumanUsersWithTargetTag: Seq[HumanUser], janusUsernames: Seq[String]): Seq[VulnerableUser] =
     iamHumanUsersWithTargetTag.filterNot { iamUser =>
-      val maybeTag = iamUser.tags.find(tag => tag.key == targetKey)
+      val maybeTag = iamUser.tags.find(tag => tag.key == USERNAME_TAG_KEY)
       maybeTag match {
         case Some(tag) => janusUsernames.contains(tag.value) // filter out human users that have tags which match the janus usernames
         case None => true
       }
```
Comment (Nirvikalpa108) [inline hq/app/schedule/unrecognised/IamUnrecognisedUsers.scala:18]:
```suggestion
      iamUser.tags
        .find(tag => tag.key == USERNAME_TAG_KEY)
        .exists(tag => janusUsernames.contains(tag.value))
```

**Review by Nirvikalpa108 — APPROVED** [review]

File: hq/app/schedule/unrecognised/IamUnrecognisedUsers.scala (hunk 4 above)
Comment (jfsoul) [reply hq/app/schedule/unrecognised/IamUnrecognisedUsers.scala:18]: Does that work if the tag doesn't exist (where we want the expression to return true)? Either way, we can update this code in another PR if worthwhile.

**Pull request merged**